src/dotty/tools/dotc/repl/CompilingInterpreter.scala

```diff
@@ -662,29 +663,68 @@ class CompilingInterpreter(
 
     private class GenericHandler(statement: Tree) extends StatementHandler(statement)
 
-    private class ValHandler(statement: ValDef) extends StatementHandler(statement) {
-      override val boundNames = List(statement.name)
+    private abstract class ValOrPatHandler(statement: Tree)
+        extends StatementHandler(statement) {
+      override val boundNames: List[Name] = _boundNames
       override def valAndVarNames = boundNames
 
       override def resultExtractionCode(req: Request, code: PrintWriter): Unit = {
-        val vname = statement.name
-        if (!statement.mods.is(Flags.AccessFlags) &&
-          !(isGeneratedVarName(vname.toString) &&
-            req.typeOf(vname.encode) == "Unit")) {
-          val prettyName = vname.decode
-          code.print(" + \"" + prettyName + ": " +
-            string2code(req.typeOf(vname)) +
-            " = \" + " +
-            " (if(" +
-            req.fullPath(vname) +
-            ".asInstanceOf[AnyRef] != null) " +
-            " ((if(" +
-            req.fullPath(vname) +
-            ".toString().contains('\\n')) " +
-            " \"\\n\" else \"\") + " +
-            req.fullPath(vname) + ".toString() + \"\\n\") else \"null\\n\") ")
-        }
+        if (!shouldShowResult(req)) return
+
+        val resultExtractors =
+          for (varName <- boundNames)
+            yield resultExtractionCode(req, varName)
```

Comment:
`val resultExtractors = boundNames.map(name => resultExtractionCode(req, name))` is more concise